src/users/users.controller.ts

```diff
@@ -298,6 +301,17 @@ export class UsersController {
     )
     dto: CreateUserDto,
   ): Promise<User> {
+    const isRecaptchaValid = await this.recaptchaVerificationService.verify(
+      dto.recaptcha,
+    );
+
+    if (!isRecaptchaValid) {
+      throw new HttpException(
+        'ReCAPTCHA verification failed',
+        HttpStatus.UNAUTHORIZED,
```

Comment:
Don't use unauthorized for this, return validation failed. Unauthorized is for when things like passport auth fail.